docs/index.md

```diff
@@ -0,0 +1,332 @@
+that uses timing related web APIs.
+
+For example, use **lolex** to simulate a system clock in your testing environment to control
+the flow of time when your testing scenario requires scheduling semantics.
+
+**lolex** implemnts the following methods:
+
+* [`setTimeout`](#settimeoutcallback-timeout-args)
+* [`clearTimeout`](#cleartimeoutid)
+* [`setImmediate`](#setimmediatecallback-args)
+* [`clearImmediate`](#clearimmediateid)
+* [`setInterval`](#setintervalcallback-timeout-args)
+* [`clearInterval`](#clearintervalid)
+* [`requestAnimationFrame`](#requestanimationframecallback)
+* [`cancelAnimationFrame`](#cancelanimationframeid)
+* [`Date`](#date)
+* [`nextTick`](#nexttickcallback-args)
+* [`queueMicrotask`](#queuemicrotaskcallback)
+* [`performance.now`](#performancenow)
+
+
+## Installation
+
+To install **lolex**, enter the following into the command line:
+
+```shell
+npm install lolex --save-dev
+```
+
+Alternately, you can [build the package yourself](https://github.com/sinonjs/lolex/blob/master/package.json#L22).
+
+
+## Usage
+
+**lolex** can be used in both Node.js and browser environments.
+
+In addition in browser environment **lolex** provides a [`performance`](#performancenow) implementation that gets its time from the clock.
+In Node.js environments **lolex** provides a [`nextTick`](#nexttick) implementation that is synchronized with the clock - and a [`process.hrtime`](#hrtimeprevtime) shim that works with the clock.
+
+If you want to use **lolex** in a browser you can use
+[the pre-built version](https://github.com/sinonjs/lolex/blob/master/lolex.js)
+available in the repo and the npm package.
+
+Using `unpkg` you only need to reference **lolex** in your `<script>` tags.
+
+```html
+<script type="text/javascript" src="https://unpkg.com/lolex/lolex.js">
+...
+</script>
+```
+
+The following code sample creates a new clock, schedules an event with the
+[`setTimeout` ](#setTimeout) function, and moves the clock ahead using the
+[`tick`](#tick) method:
+
+```js
+// In the browser distribution, a global lolex is already available
+var lolex = require("lolex");
+
+var clock = lolex.createClock();
+
+clock.setTimeout(function () {
+    console.log("The poblano is a mild chili pepper originating in the state of Puebla, Mexico.");
+}, 15);
+
+// ...
+
+clock.tick(15);
+```
+
+**NOTE:**  In the browser distribution, a global **lolex** clock is available by default.
+
+After the last line is executed, an interesting fact about the [Poblano](http://en.wikipedia.org/wiki/Poblano)
+pepper is displayed on the screen.
+If you want to simulate asynchronous behavior, you have to use your imagination when calling the various functions.
+
+In addition to the [`tick`](#tick) method, the [`next`](#next), [`runAll`](#runall), [`runToFrame`](#runtoframe),
+and [`runToLast`](#runtolast) methods are available to advance the clock.
+See the [`clock API Reference`](#clock-api) for more details.
+
+
```

Comment:
Here are my changes for the next 2 sections, along with a few questions/suggestions:

- I think we should remove the comments from the code samples - I think they clutter the samples up too much. I would rather add a sentence before or after explaining how the code is executed than have notes in the samples.  Thoughts?

- When you say "a global lolex is available by default, do you just mean "initiate lolex in the global context"? so could i replace this:

// In the browser distribution, a global lolex clock is available by default

with this:

// Initiate lolex in the global context

???

I will try to knock out the rest of the doc this weekend while i have some free time. 
********************************************************************************************

### Faking native timers

To test timers with **lolex**, replace your native timers so that calling the `setTimeout` method schedules a callback with your clock instance, not within the browser's internal clock.

Use the [`install`](#installconfig) method with no arguments to replace your native timers, and then restore `setTimeout` to it's native timers with `uninstall`:

```js
// In the browser distribution, a global lolex clock is available by default
var lolex = require("lolex");
var clock = lolex.install();
// Equivalent to:
// var clock = lolex.install(typeof global !== "undefined" ? global : window);
setTimeout(fn, 15); // Schedules with clock.setTimeout

clock.tick(15); // _fn_ argument is called

clock.uninstall();
```

To fake timers in a specific context, pass the context as an argument into the `install` method:

```js
var lolex = require("lolex");
var context = {
   setTimeout: setTimeout // context.setTimeout uses the global setTimeout method by default
}
var clock = lolex.install({target: context});
context.setTimeout(fn, 15); // Schedules with clock.setTimeout
clock.uninstall(); // Restore setTimeout to the original implementation

```
As a best practice, install the fake timers in the global object by calling [`install`](#installconfig) without arguments.

#### Automatically increment mocked time

In **lolex** 2.0 and later, you can attach a fake timer to simulate a change in the real system time, so you no longer need to predict when to use the `tick()` method.

This is accomplished by using the `setImmediate()` method at a specified interval using the `config.advanceTimeDelta`.
**NOTE:** The default value for `advanceTimeDelta` is 20ms. This increments the clock every 20ms instead of real time.

The following example sets `advanceTimeDelta` to 40ms and passes it as an argument in the `intsall` method:

```js
var lolex = require("lolex");
 var clock = lolex.install({shouldAdvanceTime: true, advanceTimeDelta: 40});
 setTimeout(() => {
    console.log('this just timed out'); //executed after 40ms
}, 30);
 setImmediate(() => {
    console.log('not so immediate'); //executed after 40ms
});
 setTimeout(() => {
    console.log('this timed out after'); //executed after 80ms
    clock.uninstall();
}, 50);
```